cmd/rook/ceph/cleanup.go

```diff
@@ -58,21 +62,63 @@ func startCleanUp(cmd *cobra.Command, args []string) error {
 			logger.Infof("successfully cleaned up %q directory", cleanupDirPath)
 		}
 		// Remove all the mon directories.
-		monDirs, err := filepath.Glob(path.Join(dataDirHostPath, "mon-*"))
+		if monSecret == "" {
+			return errors.Errorf("failed to clean up mon directories in the dataDirHostPath %q due to empty monitor secret", dataDirHostPath)
+		}
+
+		return cleanMonDir()
```

Comment:
Instead of returning an error, let's log it, then return nil. Otherwise the job will keep retrying. A one-time best-effort cleanup seems sufficient.